src/containers/Signup/Signup.js

```diff
@@ -0,0 +1,183 @@
+import React from "react";
+import { withRouter } from "react-router-dom"
+
+import Input from "../../components/Input/Input";
+import { connect } from "react-redux";
+import { ShowMessage } from "../../redux/actions/SnackBarAction";
+import { apiCalls } from "../../Helpers/axios";
+import { validate } from "../../Helpers/inputValidation";
+
+export class UnconnectedSignup extends React.Component {
+
+    state = {
+        user: {
+            email: "",
+            password: "",
+            username: "",
+            confirmPassword: "",
+        },
+        errors: {
+            email: "",
+            password: "",
+            username: "",
+            confirmPassword: "",
+        },
+        signup: {
+            value: "Submit",
+            disabled: "disabled"
+        }
+    }
+
+    handleChange = (event) => {
```

Comment:
What do you think of refactoring this function? I think its really long for an input `handle Change` function. I get that you are also handling validation when a user inputs data, but maybe we can split the logic into different functions. For instance, have the logic for setting errors in a different error handling function and still have it called inside this `handlechange` function. This may be awesome for reusability

Reply:
Thank you for the feedback. The code is now refactored.